scheduler-server/scheduling-dsl-compiler/.gitignore

```diff
@@ -0,0 +1,5 @@
+build/
+node_modules/
+tsconfig.tsbuildinfo
+scheduling-dsl-compiler-log.txt
```

Comment:
Lines 2 and 3 can be moved to the top-level `.gitignore`, then all git-ignore rules can be centralized and the `build/` rule won't be needed.

Also, why/when is `scheduling-dsl-compiler-log.txt` being generated? Just curious

Reply:
Will do :+1: 

scheduling-dsl-compiler-log.txt is generated whenever the `scheduling-dsl-compiler/src/main.ts` file is run - I mainly use this log file for debugging the typescript, since as a subprocess of java, its standard error doesn't get output to the console. Right now every single request and response is logged in this file - I might reduce the verbosity to just errors.

`tsconfig.tsbuildinfo` is how typescript keeps track of previously compiled files so it doesn't have to recompile the whole project every time. The project is so small that I think it would be better to disable this functionality, so this file is never created. I'll look into that.

Action items for me:
- [ ] Remove build and node_modules from this gitignore
- [ ] Remove the need for tsconfig.tsbuildinfo and remove it from this gitignore
- [ ] Reduce verbosity in scheduling-dsl-compiler-log.txt

Reply:
>I mainly use this log file for debugging the typescript, since as a subprocess of java, its standard error doesn't get output to the console.

When creating a sub-process you can redirect the stderr to the parent process's stdout/stderr. Can you point me to where this sub-process is being spawned?

Reply:
If you have ~15 mins. tomorrow it may be easiest to just walk through some of this, probably quicker than responding to all the comments I dropped :)